src/mlpack/core/optimizers/sgd/update_policies/momentum_update.hpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file momentum_update.hpp
+ * @author Arun Reddy
+ *
+ * Momentum update for SGD
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SGD_MOMENTUM_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_SGD_MOMENTUM_UPDATE_HPP
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Momentum update policy for SGD.
+ *
+ */
+class MomentumUpdate {
+ public:
+
+  /**
+   * Momentum update constructor.
+   *
+   * @param momentum The momentum hyperparameter
+   */
+  MomentumUpdate(const double momentum = 0.5):
+      momentum(momentum)
+  {/* do nothing */};
+
+  /**
+   * Intialize the velocity matrix to the zeros matrix of the given size
```

Comment:
It's probably worth pointing out in the comments here that this method is called by the optimizer at the start of optimization.